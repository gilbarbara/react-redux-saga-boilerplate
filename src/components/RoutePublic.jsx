import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';

const RoutePublic = ({ component: Component, isAuthenticated, to, ...rest }) => (
  <Route
    {...rest}
    render={props => (isAuthenticated ? <Redirect to={to} /> : <Component {...props} />)}
  />
);

RoutePublic.propTypes = {
  component: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  to: PropTypes.string,
};

RoutePublic.defaultProps = {
  to: '/private',
};

export default RoutePublic;
