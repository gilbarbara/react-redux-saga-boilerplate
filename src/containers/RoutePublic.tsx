import React from 'react';
import { Redirect, Route } from 'react-router-dom';

import { RouteProps } from 'types';

function RoutePublic({
  component: Component,
  isAuthenticated,
  to = '/private',
  ...rest
}: RouteProps) {
  return (
    <Route
      {...rest}
      render={props =>
        isAuthenticated ? <Redirect to={{ pathname: to }} /> : <Component {...props} />
      }
    />
  );
}

export default RoutePublic;
