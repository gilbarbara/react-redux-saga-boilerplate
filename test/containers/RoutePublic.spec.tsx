import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';

import RoutePublic from 'containers/RoutePublic';

import { render } from 'test-utils';

describe('RoutePublic', () => {
  it('should render the Login component for unauthenticated access', () => {
    const { container } = render(
      <Router initialEntries={['/login']}>
        <RoutePublic
          exact
          path="/login"
          component={() => <div>LOGIN</div>}
          isAuthenticated={false}
        />
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should redirect to /private for authenticated access', () => {
    const { container } = render(
      <Router initialEntries={['/login']}>
        <RoutePublic
          exact
          path="/login"
          component={() => <div>LOGIN</div>}
          isAuthenticated={true}
        />
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });
});
