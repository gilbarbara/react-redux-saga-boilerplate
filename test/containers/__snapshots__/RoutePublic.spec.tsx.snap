// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutePublic should redirect to /private for authenticated access 1`] = `
<div>
  <div>
    REDIRECT
  </div>
</div>
`;

exports[`RoutePublic should render the Login component for unauthenticated access 1`] = `
<div>
  <div>
    LOGIN
  </div>
</div>
`;
